File: app/src/main/java/com/example/myappcontacts/data/providers/database/IDataBaseProvider.java
package com.example.myappcontacts.data.providers.database;

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel;

import java.util.List;
import java.util.UUID;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface IDataBaseProvider {

    Single<UUID> addContact();

    Completable updateContact(ContactsModel contactsModel);

    Single<ContactsModel> loadContact(UUID contactId);

    Single<List<ContactsModel>> loadContactsList();

    Completable deleteContact(UUID contactId);

}

File: app/src/main/java/com/example/myappcontacts/data/providers/database/IDataBaseProvider.kt
package com.example.myappcontacts.data.providers.database

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel
import com.example.myappcontacts.data.dao.contactsmap.MapMarkersModel
import io.reactivex.Completable
import io.reactivex.Single
import java.util.*

interface IDataBaseProvider {
    fun addContact(): Single<UUID>

    fun updateContact(contactsModel: ContactsModel): Completable

    fun loadContact(contactId: UUID): Single<ContactsModel>

    fun loadContactsList(): Single<List<ContactsModel>>

    fun deleteContact(contactId: UUID): Completable

    fun loadMapMarkersList(): Single<List<MapMarkersModel>>
}